docs/guides/migration-guide-oauth.md

```diff
@@ -0,0 +1,240 @@
+---
+page_title: "Confluent Provider OAuth Authentication: Migration Guide"
```

Comment:
nit: we may want to pick a slightly different title to match with the existing guides:
<img width="1558" height="1166" alt="image" src="https://github.com/user-attachments/assets/22528e7f-17e0-41e2-89eb-718af6097f35" />